File: while.p.lua
local function isPrime(n)
	local i = 3
	while i < n / 2 do
		if n % i == 0 then
			return false
		end

		i = i + 2
	end

	return true
end

local prime = 0

local max = tonumber(arg[#arg])

local s = os.clock()

local i = 1
while i < max do
	if isPrime(i) then
		prime = prime + 1
	end

	i = i + 2
end

print((os.clock() - s) * 1000)

File: while.p.rs
use std::{env, time::Instant};

fn is_prime(n: i64) -> bool {
    let mut i = 3;
    while i < n / 2 {
        if n % i == 0 {
            return false;
        }

        i += 2;
    }

    return true;
}

fn main() {
    let max: i64 = env::args().last().map(|v| v.parse().unwrap()).unwrap();

    let start = Instant::now();
    let mut prime = 0;

    let mut i = 1;
    while i < max {
        if is_prime(i) {
            prime += 1;
        }

        i += 2;
    }

    println!("{}", start.elapsed().as_secs_f64() * 1000.);
    drop(prime);
}
